src/chainstate/stacks/db/accounts.rs

```diff
@@ -474,6 +563,198 @@ impl StacksChainState {
         Ok(())
     }
 
+    /// Store a matured miner reward for subsequent query in Clarity, without doing any validation
+    fn inner_insert_matured_miner_payment<'a>(
+        tx: &mut DBTx<'a>,
+        parent_block_id: &StacksBlockId,
+        child_block_id: &StacksBlockId,
+        reward: &MinerReward,
+    ) -> Result<(), Error> {
+        // the only time it's okay to re-insert the same reward is if there are two Stacks forks
+        // trying to store the same matured rewards for a common ancestor block.
+        let cur_rewards = StacksChainState::inner_get_matured_miner_payments(
+            tx,
+            parent_block_id,
+            child_block_id,
+        )?;
+        if cur_rewards.len() > 0 {
+            let mut present = false;
+            for rw in cur_rewards.iter() {
+                if (rw.is_parent() && reward.is_parent()) || (rw.is_child() && reward.is_child()) {
+                    // must insert a parent or a child at most once
+                    assert_eq!(rw, reward, "FATAL: tried to insert multiple distinct matured parent block reward records");
+                    present = true;
+                }
+            }
+
+            if present {
+                return Ok(());
+            }
+        }
+
+        // not present
+        let sql = "INSERT INTO matured_rewards (
+            recipient,
+            vtxindex,
+            coinbase,
+            tx_fees_anchored,
+            tx_fees_streamed_confirmed,
+            tx_fees_streamed_produced,
+            parent_index_block_hash,
+            child_index_block_hash
+        ) VALUES (?1,?2,?3,?4,?5,?6,?7,?8)";
+
+        let args: &[&dyn ToSql] = &[
+            &reward.address.to_string(),
+            &reward.vtxindex,
+            &format!("{}", &reward.coinbase),
+            &format!("{}", &reward.tx_fees_anchored),
+            &format!("{}", &reward.tx_fees_streamed_confirmed),
+            &format!("{}", &reward.tx_fees_streamed_produced),
```

Comment:
`x.to_string()` is always the same as `format!("{}", x)`

```suggestion
            &reward.coinbase.to_string(),
            &reward.tx_fees_anchored.to_string(),
            &reward.tx_fees_streamed_confirmed.to_string(),
            &reward.tx_fees_streamed_produced.to_string(),
```